src/components/TodoItem.tsx

```diff
@@ -0,0 +1,113 @@
+import React, { useContext, useState } from 'react';
+import { Todo } from '../types/Todo';
+import classNames from 'classnames';
+import { TodosContext } from './TodosContext';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const { todos, setTodos } = useContext(TodosContext);
+  const [editingTodoId, setEditingTodoId] = useState<number | null>(null);
+  const [newTitle, setNewTitle] = useState(todo.title);
+
+  function handleClickDeleteTodo(TodoId: number) {
+    const newTodos = todos.filter(currTodo => currTodo.id !== TodoId);
+
+    setTodos(newTodos);
+  }
+
+  const handleEditTodo = (TodoChange: Todo) => {
+    let newTodos = todos;
+
+    if (newTitle === TodoChange.title) {
+      setEditingTodoId(null);
+
+      return;
+    }
+
+    if (newTitle.trim()) {
+      newTodos = todos.map(curr => {
+        if (curr.id === todo.id) {
+          return {
+            ...curr,
+            title: newTitle,
+          };
+        }
+
+        return curr;
+      });
+    } else {
+      handleClickDeleteTodo(todo.id);
+
+      return;
+    }
+
+    setTodos(newTodos);
+    setEditingTodoId(null);
+  };
+
+  const handleChangeChecked = (todoChange: Todo) => {
+    const newTodos = todos.map(currTodo => {
+      if (currTodo.id === todoChange.id) {
+        return {
+          ...currTodo,
+          completed: !todoChange.completed,
+        };
+      }
+
+      return currTodo;
+    });
+
+    setTodos(newTodos);
+  };
+
+  return (
+    <li
+      data-id={todo.id}
+      className={classNames({
+        completed: todo.completed,
+        editing: todo.id === editingTodoId,
+      })}
+      onDoubleClick={() => setEditingTodoId(todo.id)}
+    >
+      <div className="view">
+        <input
+          type="checkbox"
+          className="toggle"
+          id="toggle-view"
+          checked={todo.completed}
+          onChange={() => handleChangeChecked(todo)}
+        />
+        <label>{todo.title}</label>
+        <button
+          type="button"
+          className="destroy"
+          data-cy="deleteTodo"
+          onClick={() => handleClickDeleteTodo(todo.id)}
+        />
+      </div>
+      {todo.id === editingTodoId && (
```

Comment:
FYI: since it's inside one TodoItem - id will always be equal to this `todo.id` or `null`. I.e. there is only 2 possible values - you can use boolean state for it instead of `editingTodoId`